### DISCUSSION TITLE
Are there plans for an admin panel for the PDS?

### DISCUSSION THREAD

**Discussion by apppel** [bluesky-social/atproto#1721] [Protocol (atproto)]
Are there plans for an admin panel for the PDS? Haven't setup the PDS myself yet, but looking in to it right now. Would make sense for users to manage the PDS as simple as possible. 

Have searched for it, but couldn't find anything. Thanks!

**Comment by bnewbold** [comment]
Good question! The short answer is that there isn't a specific web admin interface yet. Bluesky will probably build somebody if nothing takes off from self-hosting community.

The longer answer is that "where moderation happens" has shifted around a couple times. There is a need for PDS admins to do at least basic moderation in the form of account takedowns. Bluesky (the company) currently primarily does moderation at the AppView, and propagates takedown decisions back to the PDS instances, so moderation needs at the PDS is very minimal for us (just an API). We are also thinking of breaking out the moderation service from the AppView into a separate service, which would allow folks who don't run a PDS *or* an AppView to participate (eg, via third-party labels). Because these roles are still shifting around, there isn't a clear use-case of what to build quite yet. Eg, should there be one unified admin interface that *can* work with everything? Separate interfaces for each piece of software? Etc.

**Reply by apppel** [reply]
Thanks for the answer, that clarifies it a bit ;) I will look into setting up my PDS right now and explore use cases for an admin (and also stats) panel. 

Also: is it the intention that the user manages it's repository by logging into the PDS? Or do you prefer to keep that in the AppView?

**Reply by bnewbold** [reply]
The PDS controls repository on behalf of the account holder (user). The PDS generally holds the signing key for the account, which is required to update (write to) the repo. Other services and clients all send requests to the PDS which does the actual updates.

**Reply by apppel** [reply]
Hi Brian,

Thanks, yes that was clear to me. My question more specifically: it seems to me that a user can have *one* account that is used by multiple AppView services (e.g. BlueSky and BlueReddit). It then seems logical to me that the user can manage access to these accounts on the PDS level, and revoke access to the repository by a specific service. 

For example if i delete one of my accounts on the Sandbox right now, both the account and the did are deleted (see screenshot, second account gets deleted). It seems to me that this is not desired as that same account should also be usable in other services (BlueReddit).

Do I understand the account logic incorrectly or is this account management indeed something that should be added? Thanks!


<img width="1429" alt="Screenshot 2023-10-11 at 09 27 42" src="https://github.com/bluesky-social/atproto/assets/54106191/d87a2ca9-1067-447c-a8b7-f0c4573d0564">

**Reply by apppel** [reply]
@bnewbold Is it possible for you or someone from the team to get back on my last question? I have searched through all the docs and all I could find was this:

> **3rd Party Authentication and Authorization**: the intention is to provide a more powerful auth system, both for authentication across services (within atproto), and for "log in with atproto" use cases outside of the ecosystem. This will likely incorporate existing auth standards.

If this is indeed the intention, it seems to me logical that the user can log in to a PDS and manage authentication access for the different services (including an overview of all the services using the at proto account). Looking forward to reading what you think, because this is something I would be interested in contributing to.

Thanks in advance!

**Reply by bnewbold** [reply]
Quick notes on your question:

> revoke access to the repository by a specific service

This isn't a planned core functionality. All repository content in the network is considered public, without permission or access control. Similar to the public web. There may be rate-limits or other operational limits, but generally the escalations to remove extreme bad actors are around filtering out or simply not consuming their content in to aggregated views of the network.

A loose sketch of what service-to-service auth would look like is signed bearer tokens with authn attenuations, similar to UCANs, or the older Macaroons or even Kerberos systems. The PDS would generate these tokens and staple them to outgoing requests (eg, to an AppView or Feed Generator) as needed. The receiving service could authenticate the request using only public identity info (eg, DID document). There would not be explicit in-band account management, though service providers could implement logins and UIs to manage accounts if they want, or ban/block service to bad actors under their own policies.

This is something we want to come up with a proposal/plan on internally as a first step.
